console/console-init/ui/src/modules/iot-device/components/AddGateways/AddGateway.tsx

```diff
@@ -52,54 +98,142 @@ export const AddGateways: React.FunctionComponent<IGatewaysProps> = ({
     setGateways([...gateways]);
   };
 
-  const alertDescription = `In AMQ IoT, any existing device directly connected to cloud can turn
-    to be a gateway by been added to other devices for their connections. Multiple gateways are supported.`;
+  const removeGatewayGroup = (group: string) => {
+    const idIndex = gatewayGroups.indexOf(group);
+    gatewayGroups.splice(idIndex, 1);
+    setGatewayGroups([...gatewayGroups]);
+  };
+
+  const onTypeSelect = (val: string) => {
+    setInputType(val);
+    setSelectedGateways([]);
+    setInputID("");
+  };
+
+  const deviceIDValidationState = () => {
+    if (inputID && inputID.length > 1 && !isValid)
+      return ValidatedOptions.error;
+    return ValidatedOptions.default;
+  };
+
+  const disableAddGatewayBtn = () => {
+    if (
+      selectedGateways?.length > 0 ||
+      (inputID && inputID?.trim() !== "" && isValid)
+    ) {
+      return false;
+    }
+    return true;
+  };
+
+  const onSelect = (_: any, selection: any) => {
```

Comment:
suggestion to do same thing in different way. follow it if works for you.

  const onSelect = (_: any, selection: any) => {
    const newSelectedGateways=[...selectedGateways, selection];
    if (selectedGateways.includes(selection)) {
        newSelectedGateways=selectedGateways.filter(item => item !== selection);   
    } 
    setSelectedGateways(newSelectedGateways);
  };